background/loadScripts.js

```diff
@@ -43,36 +44,52 @@ export default function loadScripts (port) {
     }
 
     if (!codeCache) {
-      const vendor = fetch(`http://${UI}/vendor.js`)
-        .then(checkResponseOk)
-        .then(response => response.blob());
-      const embed = fetch(`http://${UI}/embed.html`)
+      codeCacheVersion = State.version;
+
+      codeCache = fetch(`http://${UI}/embed.json`)
         .then(checkResponseOk)
-        .then(response => response.text())
-        .then(page => ({
-          styles: /styles\/embed\.([a-z0-9]{10})\.css/.exec(page),
-          scripts: /embed\.([a-z0-9]{10})\.js/.exec(page)
-        }))
-        .then(res => {
-          return Promise.all([
-            fetch(`http://${UI}/${res.styles[0]}`).then(checkResponseOk),
-            fetch(`http://${UI}/${res.scripts[0]}`).then(checkResponseOk)
-          ]);
+        .then((response) => response.json())
+        .then((embed) => {
+          const { assets } = embed;
+
+          const filteredAssets = assets
+            .filter((asset) => !/html$/.test(asset))
+            .filter((asset) => !/^embed(.+)js$/.test(asset));
+
+          const mainScript = assets.find((asset) => /^embed(.+)js$/.test(asset));
+
+          const assetsPromises = filteredAssets
+            .map((asset) => {
+              return fetch(`http://${UI}/${asset}`)
+                .then((response) => response.blob())
+                .then((blob) => {
+                  if (/\.js$/.test(asset)) {
+                    return URL.createObjectURL(blob, { type: 'application/javascript' });
+                  }
+
+                  return URL.createObjectURL(blob);
+                })
+                .then((url) => ({ path: asset, url }));
+            });
+
+          const scriptPromise = fetch(`http://${UI}/${mainScript}`)
+            .then((response) => response.text());
+
+          return Promise.all([ scriptPromise, Promise.all(assetsPromises) ]);
         })
-        .then(responses => Promise.all(responses.map(response => response.blob())));
+        .then(([ script, assets ]) => {
+          assets.forEach((asset) => {
```

Comment:
It's true, I actually thought of that, but was easier this way. Would actually be better. However I dont see why it wouldn't work? It replaces `image.src = "/ethereum.png"` with `image.src = "blob:chrome-extension://<extention-id>/<hash-url>"`, no ? So it could be loaded from anywhere within the Chrome context